stl/inc/chrono

```diff
@@ -4613,8 +4613,10 @@ namespace chrono {
             const _Ctype& _Ctype_fac{_STD use_facet<_Ctype>(_Iosbase.getloc())};
             constexpr _InIt _Last{};
 
-            while (*_Fmt != '\0' && _First != _Last && _State == ios_base::goodbit) {
-                if (*_Fmt == '%') {
+            while (*_Fmt != '\0' && (_State & ~ios_base::eofbit) == ios_base::goodbit) {
+                if (_First == _Last) {
+                    _State |= ios_base::failbit | ios_base::eofbit;

```

Comment:
After looking at this control flow, I believe that this is correct but complicated to keep track of - we end up executing `++_Fmt;` (which is fortunately safe and non-observable) before `return _First;`. I think it would be simpler to directly add a `return _First;` after this line - even though it's redundant, it's simpler to see that we immediately return after setting `failbit` and `eofbit`.

Reply:
Agreed, although I prefer `break` to match `_Time_parse_fields` below - it feels more defensive